### ISSUE TITLE
how about dubbo?

### ISSUE THREAD

**Issue by walter211** [line/armeria#1759]

**Comment by trustin** [comment]
Hi @walter211, are you asking for Dubbo protocol support in Armeria? It'd be nice if you could add more details, like updating the issue description.

**Comment by eonezhang** [comment]
as I know, dubbo is a framework based on tcp, you can get more detail from [Implementation details](http://dubbo.apache.org/en-us/docs/dev/implementation.html)

**Comment by trustin** [comment]
Yeah, I wonder if he is asking for Dubbo RPC protocol support or he is looking for the comparison between Dubbo and Armeria.

**Comment by walter211** [comment]
> Yeah, I wonder if he is asking for Dubbo RPC protocol support or he is looking for the comparison between Dubbo and Armeria.

Yes!! thats exactly what I mean, cause I am tring to refactor a java project to rust. It seems you guys is the best team to make this work.

**Comment by trustin** [comment]
>> Yeah, I wonder if he is asking for Dubbo RPC protocol support or he is looking for the comparison between Dubbo and Armeria.
>
> Yes!! thats exactly what I mean, cause I am tring to refactor a java project to rust. It seems you guys is the best team to make this work.

Which one? :sweat_smile: I guess you asked for the former, right?

**Comment by trustin** [comment]
It seems like Dubbo RPC protocol is not HTTP-based, right?

**Comment by walter211** [comment]
> It seems like Dubbo RPC protocol is not HTTP-based, right?

right

**Comment by trustin** [comment]
That being said, it will take non-trivial effort to add support for non-HTTP protocol because Armeria's API assumes HTTP in many places. It is not impossible though.

**Comment by eonezhang** [comment]
> That being said, it will take non-trivial effort to add support for non-HTTP protocol because Armeria's API assumes HTTP in many places. It is not impossible though.

It seems dubbo supporting is a long way to go, although most of our systems were build on top dubbo, I've never think of armeria can play the role for dubbo, we just extends `dubbo` to support `armeria` by the extension point [Protocol](http://dubbo.apache.org/en-us/docs/dev/impls/protocol.html)
